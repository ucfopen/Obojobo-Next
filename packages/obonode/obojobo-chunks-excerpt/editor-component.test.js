import React from 'react'
import renderer from 'react-test-renderer'

import Code from './editor-component'

jest.mock('slate')
jest.mock('slate-react')
jest.mock(
	'obojobo-document-engine/src/scripts/oboeditor/components/node/with-slate-wrapper',
	() => item => item
)
jest.mock(
	'obojobo-document-engine/src/scripts/oboeditor/components/node/editor-component',
	() => props => <div>{props.children}</div>
)

describe('Code Editor Node', () => {
	test('Code builds the expected component', () => {
		const component = renderer.create(
			<Code
				node={{
					data: {
						get: () => {
							return {}
						}
					},
					content: {}
				}}
				parent={{
					getPath: () => ({
						get: () => 0
					}),
					nodes: {
						size: 2
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
