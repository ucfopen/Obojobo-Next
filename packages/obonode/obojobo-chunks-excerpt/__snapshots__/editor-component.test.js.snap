// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Editor Node Code builds the expected component 1`] = `
<div>
  <div
    className="text-chunk obojobo-draft--chunks--code pad"
  >
    <pre>
      <code />
    </pre>
  </div>
</div>
`;
